qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -0,0 +1,126 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A collections of libraries to setup Calibrations.
+
+Note that the set of available functions will be extended in future releases.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Tuple
+
+from qiskit.circuit import Parameter
+import qiskit.pulse as pulse
+from qiskit.pulse import ScheduleBlock
+
+from qiskit_experiments.calibration_management.calibration_key_types import ParameterValueType
+
+
+class BasisGateLibrary(ABC):
+    """A base class for libraries of basis gates to make it easier to setup Calibrations."""
+
+    # Location where default parameter values are stored. These may be updated at construction.
+    __default_values__ = dict()
+
+    def __init__(self, default_values: Optional[Dict] = None):
+        """Setup the library.
+
+        Args:
+            default_values: A dictionary to override library default parameter values.
+        """
+        self._schedules = dict()
+
+        if default_values is not None:
+            for key, value in default_values.items():
+                self.__default_values__[key] = value
+
+    def __getitem__(self, name: str) -> ScheduleBlock:
+        """Return the schedule."""
+        return self._schedules[name]
+
+    @abstractmethod
+    def default_values(self) -> List[Tuple[ParameterValueType, Parameter, Tuple, ScheduleBlock]]:
+        """Return the default values for the parameters.
+
+        Returns
+            A list of tuples is returned. These tuples are structured so that instances of
+            :class:`Calibrations` can call :meth:`add_parameter_value` on the tuples.
+        """
+
+
+class FixedFrequencyTransmonSXRZ(BasisGateLibrary):
+    """A library of gates for fixed-frequency superconducting qubit architectures."""
+
+    __default_values__ = {"duration": 160, "amp": 0.5, "β": 0.0}
+
+    def __init__(self, default_values: Optional[Dict] = None):
+        """Setup the schedules."""
+        super().__init__(default_values)
+
+        dur = Parameter("duration")
+        sigma = Parameter("σ")
+
+        # X gates
+        sched_x = self._single_qubit_schedule("x", dur, Parameter("amp"), sigma, Parameter("β"))
+
+        # square-root X gates
+        sched_sx = self._single_qubit_schedule("sx", dur, Parameter("amp"), sigma, Parameter("β"))
+
+        for sched in [sched_x, sched_sx]:
+            self._schedules[sched.name] = sched
+
+    @staticmethod
+    def _single_qubit_schedule(
+        name: str,
+        dur: Parameter,
+        amp: Parameter,
+        sigma: Parameter,
+        beta: Optional[Parameter] = None,
+    ) -> ScheduleBlock:
+        """Build a single qubit pulse."""
+
+        chan = pulse.DriveChannel(Parameter("ch0"))
+
+        if beta is not None:
```

Comment:
why we cannot initialize DRAG with beta=0?

Reply:
The implementation there was a bit clumsy. This looks better: 4ae1763. The idea is to give users the ability to turn DRAG off.